File: Data Structures & Algorithms/validate-parentheses/submission-0.py
class Solution:
    def isValid(self, s: str) -> bool:
        stack = []
        mapping = {']':'[',')':'(','}':'{'}
        for c in s:
            if c in mapping:
                top = stack.pop() if stack else  '#'
                if top != mapping[c]:
                    return False
            else:
                stack.append(c)
        return not stack

File: Data Structures & Algorithms/decode-string/submission-0.cpp
class Solution {
public:
    string decodeString(string s) {
        stack<int> numStack;
        stack<string> strStack;
        int curNum = 0;
        string curString = "";
        for(char c : s){
            if(isdigit(c)){
                curNum = curNum * 10 + (c - '0');
            }else if(c == '['){
                numStack.push(curNum);
                strStack.push(curString);
                curNum = 0;
                curString = "";
            }else if(c == ']'){
                int repeat = numStack.top();
                numStack.pop();
                string prevString = strStack.top();
                strStack.pop();
                for(int i = 0; i < repeat; i++){
                    prevString += curString;
                }
                curString = prevString;
            }else{
                curString += c;
            }
        }
        return curString;
    }
};